libs/sdk-ui-ext/src/internal/components/dialogs/embedInsightDialog/EmbedInsightDialog.tsx

```diff
@@ -50,21 +55,19 @@ export const EmbedInsightDialog: React.VFC<IEmbedInsightDialogProps> = (props) =
         settings,
         colorPalette,
         executionConfig,
-        codeType,
-        integrationDocLink,
+        reactIntegrationDocLink,
+        webComponentIntegrationDocLink,
+        openSaveInsightDialog,
+        workspaceId,
         onClose,
         onCopyCode,
     } = props;
-    const [codeLang, setCodeLang] = useState<CodeLanguageType>("ts");
-    const [codeOption, setCodeOption] = useState<CodeOptionType>(getDefaultEmbedCodeOptions(codeType));
+    const [currentTab, setCurrentTab] = useState<EmbedType>("react");
```

Comment:
This component is going to be large pls extract all hooks into useEmbedInsightDialog(props)